presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1155,173 +1163,99 @@ public void testExpressionConstantFolding()
-        testSimpleComparison(lessThan(C_BIGINT, doubleLiteral(-2.0)), C_BIGINT, Range.lessThan(BIGINT, -2L));
-        testSimpleComparison(lessThan(C_BIGINT, doubleLiteral(0x1p64)), C_BIGINT, Range.lessThanOrEqual(BIGINT, Long.MAX_VALUE));
-        testSimpleComparison(lessThan(C_BIGINT, doubleLiteral(-0x1p64)), C_BIGINT, Range.lessThan(BIGINT, Long.MIN_VALUE));
-
-        // equal
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(2.5)), C_BIGINT, Domain.none(BIGINT));
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(2.0)), C_BIGINT, Range.equal(BIGINT, 2L));
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(-2.5)), C_BIGINT, Domain.none(BIGINT));
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(-2.0)), C_BIGINT, Range.equal(BIGINT, -2L));
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(0x1p64)), C_BIGINT, Domain.none(BIGINT));
-        testSimpleComparison(equal(C_BIGINT, doubleLiteral(-0x1p64)), C_BIGINT, Domain.none(BIGINT));
-
-        // not equal
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(2.5)), C_BIGINT, Domain.notNull(BIGINT));
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(2.0)), C_BIGINT, Domain.create(ValueSet.ofRanges(Range.lessThan(BIGINT, 2L), Range.greaterThan(BIGINT, 2L)), false));
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(-2.5)), C_BIGINT, Domain.notNull(BIGINT));
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(-2.0)), C_BIGINT, Domain.create(ValueSet.ofRanges(Range.lessThan(BIGINT, -2L), Range.greaterThan(BIGINT, -2L)), false));
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(0x1p64)), C_BIGINT, Domain.notNull(BIGINT));
-        testSimpleComparison(notEqual(C_BIGINT, doubleLiteral(-0x1p64)), C_BIGINT, Domain.notNull(BIGINT));
-
-        // is distinct from
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(2.5)), C_BIGINT, Domain.all(BIGINT));
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(2.0)), C_BIGINT, Domain.create(ValueSet.ofRanges(Range.lessThan(BIGINT, 2L), Range.greaterThan(BIGINT, 2L)), true));
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(-2.5)), C_BIGINT, Domain.all(BIGINT));
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(-2.0)), C_BIGINT, Domain.create(ValueSet.ofRanges(Range.lessThan(BIGINT, -2L), Range.greaterThan(BIGINT, -2L)), true));
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(0x1p64)), C_BIGINT, Domain.all(BIGINT));
-        testSimpleComparison(isDistinctFrom(C_BIGINT, doubleLiteral(-0x1p64)), C_BIGINT, Domain.all(BIGINT));
+        List<NumericValues> translationChain = ImmutableList.of(
+                new NumericValues(C_DOUBLE, (value) -> doubleLiteral((double) value), -1.0 * Double.MAX_VALUE, -22.0, -44.555678, 23.0, 44.555678, Double.MAX_VALUE),
```

Comment:
`Double.MIN_VALUE` is not a minimal negative value. Is a minimal positive value greater than zero.